File: zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/smoke/SmokeTest.java
/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Camunda License 1.0. You may not use this file
 * except in compliance with the Camunda License 1.0.
 */
package io.camunda.zeebe.it.smoke;

import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;

/**
 * {@code @SmokeTest} is used to signal that the annotated method is smoke test case.
 *
 * <p>A smoke test is simple test case that verifies whether the system starts properly by verifying
 * health indicators and/or by performing simple API interaction.
 */
@Target({ElementType.TYPE, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
@Tag("smoke-test")
@Test
@Inherited
public @interface SmokeTest {}

File: optimize/client/e2e/cloud-tests/smokeTest.js
/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Camunda License 1.0. You may not use this file
 * except in compliance with the Camunda License 1.0.
 */

import dotenv from 'dotenv';

import config from '../config';
import * as u from '../utils';

import * as Common from '../sm-tests/Common.elements.js';
import {loginToCloud} from './login.js';
import * as e from './smokeTest.elements.js';

if (!process.env.CI) {
  dotenv.config();
}

fixture('Smoke test').page(config.collectionsEndpoint);

test('create a report from a template', async (t) => {
  await t.maximizeWindow();

  await loginToCloud(t);

  await t.click(Common.collectionsPage);
  await t.click(Common.createNewButton);
  await t.click(Common.menuOption('Report'));
  await t.click(e.emptyTemplate);
  await t.click(Common.modalConfirmButton);
  await u.save(t);
});
